lib/state_change_syncer.go

```diff
@@ -986,6 +966,47 @@ func (stateChangeSyncer *StateChangeSyncer) SyncMempoolToStateSyncer(server *Ser
 			IsMempoolTxn:     true,
 		})
 	}
+
+	// Create a copy of the event manager, assign it to this utxo view.
+	mempoolTxEventManager := *mempoolTxUtxoView.EventManager
+
+	// Reset event manager handlers
+	mempoolTxEventManager.stateSyncerOperationHandlers = nil
+	mempoolTxEventManager.stateSyncerFlushedHandlers = nil
+	mempoolTxEventManager.OnStateSyncerOperation(stateChangeSyncer._handleStateSyncerOperation)
+	mempoolTxEventManager.OnStateSyncerFlushed(stateChangeSyncer._handleStateSyncerFlush)
+
+	mempoolTxEventManager.isMempoolManager = true
+	mempoolTxUtxoView.EventManager = &mempoolTxEventManager
+
+	// Kill the snapshot so that it doesn't affect the original snapshot.
+	mempoolTxUtxoView.Snapshot = nil
+
+	server.blockchain.ChainLock.RLock()
+	mempoolTxUtxoView.TipHash = server.blockchain.bestChain[len(server.blockchain.bestChain)-1].Hash
```

Comment:
let's use the `BlockTip` function on blockchain. This will reduce changes I need to make when we get rid of the best chain in memory.
```suggestion
	mempoolTxUtxoView.TipHash = server.blockchain.BlockTip().Hash
```